Userland/Services/SystemServer/main.cpp

```diff
@@ -154,9 +173,62 @@ static ErrorOr<void> spawn_device_mapper_process()
     return {};
 }
 
+static ErrorOr<void> prepare_tmpfs_system_devicemap_directory()
+{
+    dbgln("Creating /tmp/system/devicemap directory");
+
+    TRY(Core::System::mkdir("/tmp/system/"sv, 0755));
+    TRY(Core::System::mkdir("/tmp/system/devicemap/"sv, 0755));
+    TRY(Core::System::mount(-1, "/tmp/system/devicemap/"sv, "ram"sv, MS_NOEXEC | MS_NOSUID | MS_NODEV));
+    TRY(Core::System::mkdir("/tmp/system/devicemap/nodes/"sv, 0755));
+    TRY(Core::System::mkdir("/tmp/system/devicemap/nodes/block/"sv, 0755));
+    TRY(Core::System::mkdir("/tmp/system/devicemap/nodes/char/"sv, 0755));
+
+    TRY(Core::System::mkdir("/tmp/system/devicemap/family/"sv, 0755));
+
+    {
+        auto file = TRY(Core::File::open("/sys/kernel/chardev_major_allocs"sv, Core::File::OpenMode::Read));
+        auto file_contents = TRY(file->read_until_eof());
+        auto json_result = TRY(JsonValue::from_string(file_contents));
+        auto const& json = json_result.as_array();
+        TRY(json.try_for_each([&](auto& value) -> ErrorOr<void> {
+            auto& major_number_allocation_object = value.as_object();
+            auto allocated_number = major_number_allocation_object.get_u64("allocated_number"sv).value_or(0);
+            auto path = LexicalPath("/tmp/system/devicemap/nodes/char/").append(MUST(String::number(allocated_number)));
+            TRY(Core::System::mkdir(path.string(), 0755));
+
+            auto family_name = major_number_allocation_object.get_byte_string("family_name"sv).value_or({});
+            auto symlink_path = LexicalPath("/tmp/system/devicemap/family/").append(family_name);
+            TRY(Core::System::symlink(path.string(), symlink_path.string()));
+            return {};
+        }));
+    }
+
+    {
+        auto file = TRY(Core::File::open("/sys/kernel/blockdev_major_allocs"sv, Core::File::OpenMode::Read));
+        auto file_contents = TRY(file->read_until_eof());
+        auto json_result = TRY(JsonValue::from_string(file_contents));
+        auto const& json = json_result.as_array();
+        TRY(json.try_for_each([&](auto& value) -> ErrorOr<void> {
+            auto& major_number_allocation_object = value.as_object();
+            auto allocated_number = major_number_allocation_object.get_u64("allocated_number"sv).value_or(0);
+            auto path = LexicalPath("/tmp/system/devicemap/nodes/block/").append(MUST(String::number(allocated_number)));
+            TRY(Core::System::mkdir(path.string(), 0755));
+
+            auto family_name = major_number_allocation_object.get_byte_string("family_name"sv).value_or({});
```

Comment:
Why are you using `value_or` here? If the JSON object doesn't contain the expected attributes we should skip them and/or print an error.